src/main/kotlin/blackjack/model/BlackjackGame.kt

```diff
@@ -0,0 +1,93 @@
+package blackjack.model
+
+class BlackjackGame(private val deck: CardDeck, val participants: Participants) {
+    fun playRound(
+        askForPlayerAction: (name: ParticipantName) -> Boolean,
+        displayParticipantsStatus: (Participant) -> Unit,
+    ) {
+        playRoundForPlayers(askForPlayerAction, displayParticipantsStatus)
+        playRoundForDealer(displayParticipantsStatus)
+    }
+
+    private fun playRoundForPlayers(
+        askForPlayerAction: (name: ParticipantName) -> Boolean,
+        displayParticipantsStatus: (player: Player) -> Unit,
+    ) {
+        participants.players.forEach { player ->
+            while (player.state is Running) {
+                val continuePlaying = askForPlayerAction(player.name)
+
+                if (continuePlaying) {
+                    player.receiveCard(deck.pick())
+                } else {
+                    player.finishRound()
+                }
+                displayParticipantsStatus(player)
+            }
+        }
+    }
+
+    private fun playRoundForDealer(displayParticipantsStatus: (dealer: Dealer) -> Unit) {
+        val dealer = participants.dealer
+        while (dealer.state is Running) {
+            val continuePlaying = dealer.isUnderHitThreshold()
+            if (continuePlaying) {
+                dealer.receiveCard(deck.pick())
+            } else {
+                dealer.finishRound()
+            }
+            displayParticipantsStatus(dealer)
+        }
+    }
+
+    fun calculateResult(): Map<Participant, WinningState> {
+        val (dealer, players) = participants
+        val outcomes = getPlayersWinningState(dealer, players)
+
+        val dealerWinCount = outcomes.values.count { it.wins == 0 && it.losses == 1 }
```

Comment:
현재의 조건식으로 딜러의 승리/패배를 판단하기에는 버그를 유발할 가능성이 있어보여요.
예를들어 만약 아래와 같은 상태라면, 어떻게 카운트가 되어야할까요?
```suggestion
        val dealerWinCount = outcomes.values.count { it.wins == 1 && it.losses == 2 }
```